cmd/snap-confine/snap-device-helper-test.c

```diff
@@ -188,17 +191,21 @@ static void test_sdh_err(void)
 }
 
 static struct sdh_test_data add_data =
-    { "add", "snap.foo.bar", "devices.allow", "devices.deny" };
+    { "add", "snap.foo.bar", "snap_foo_bar", "devices.allow", "devices.deny" };
 static struct sdh_test_data change_data =
-    { "change", "snap.foo.bar", "devices.allow", "devices.deny" };
+    { "change", "snap.foo.bar", "snap_foo_bar", "devices.allow", "devices.deny" };
 static struct sdh_test_data remove_data =
-    { "remove", "snap.foo.bar", "devices.deny", "devices.allow" };
+    { "remove", "snap.foo.bar", "snap_foo_bar", "devices.deny", "devices.allow" };
 static struct sdh_test_data instance_add_data =
-    { "add", "snap.foo_bar.baz", "devices.allow", "devices.deny" };
+    { "add", "snap.foo_bar.baz", "snap_foo_bar_baz", "devices.allow", "devices.deny" };
 static struct sdh_test_data instance_change_data =
-    { "change", "snap.foo_bar.baz", "devices.allow", "devices.deny" };
+    { "change", "snap.foo_bar.baz", "snap_foo_bar_baz", "devices.allow", "devices.deny" };
 static struct sdh_test_data instance_remove_data =
-    { "remove", "snap.foo_bar.baz", "devices.deny", "devices.allow" };
+    { "remove", "snap.foo_bar.baz", "snap_foo_bar_baz", "devices.deny", "devices.allow" };
+static struct sdh_test_data add_hook_data =
+    { "add", "snap.foo.hook.configure", "snap_foo_hook_configure", "devices.allow", "devices.deny" };
+static struct sdh_test_data instance_add_hook_data =
+    { "add", "snap.foo_bar.hook.configure", "snap_foo_bar_hook_configure", "devices.allow", "devices.deny" };
```

Comment:
Perhaps add a test for 'snap.foo_hook.hook.configure'

Reply:
I think the problem is with an instance "hook" and a app "configure": `snap.foo_hook_configure` - or am I missing something?

Reply:
I mean, YES to the suggestion to add a test :) and yet it seems like there is this case (above) where the current transformation is ambiguous - unless I miss something of course.

Reply:
IMO a hook on a `foo_hook` instance is more ambiguous. With a workaround proposed by @jdstrand SNAPAPP expands correctly to `snap.foo_hook.hook.configure`, without it I get `snap.foo_hook_hook.hook.hook.hook_configure`.